### PR TITLE
Use all available processors for checks

### PR DESCRIPTION
**Pull request by Felixoid**
Changelog category (leave one):
- Build/Testing/Packaging Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Use all available in container processors for PVS studio check and fast tests. Delete coverage image

### CODE REVIEW

File: docker/test/fasttest/run.sh [hunk 1]
```diff
@@ -253,7 +253,7 @@ function run_tests
     start_server
 
     set +e
-    time clickhouse-test --hung-check -j 8 --order=random \
+    time clickhouse-test --hung-check -j "$(nproc)" --order=random \
```
Comment (alesapin) [inline docker/test/fasttest/run.sh:256]: bad idea :)

File: docker/test/pvs/Dockerfile [hunk 2]
```diff
@@ -42,7 +42,7 @@ ENV CCACHE_DIR=/test_output/ccache
 CMD echo "Running PVS version $PKG_VERSION" && mkdir -p $CCACHE_DIR && cd /repo_folder && pvs-studio-analyzer credentials $LICENCE_NAME $LICENCE_KEY -o ./licence.lic  \
     && cmake . -D"ENABLE_EMBEDDED_COMPILER"=OFF -D"DISABLE_HERMETIC_BUILD"=ON -DCMAKE_C_COMPILER=clang-13 -DCMAKE_CXX_COMPILER=clang\+\+-13 \
     && ninja re2_st clickhouse_grpc_protos \
-    && pvs-studio-analyzer analyze -o pvs-studio.log -e contrib -j 4 -l ./licence.lic; \
+    && pvs-studio-analyzer analyze -o pvs-studio.log -e contrib -j "$(nproc)" -l ./licence.lic; \
```
Comment (alesapin) [inline docker/test/pvs/Dockerfile:45]: good idea

File: docker/test/coverage/run.sh [hunk 3]
```diff
@@ -80,7 +80,7 @@ LLVM_PROFILE_FILE='client_coverage_%5m.profraw' clickhouse-client --query "RENAM
 LLVM_PROFILE_FILE='client_coverage_%5m.profraw' clickhouse-client --query "RENAME TABLE datasets.visits_v1 TO test.visits"
 LLVM_PROFILE_FILE='client_coverage_%5m.profraw' clickhouse-client --query "SHOW TABLES FROM test"
 
-LLVM_PROFILE_FILE='client_coverage_%5m.profraw' clickhouse-test -j 8 --testname --shard --zookeeper --print-time 2>&1 | ts '%Y-%m-%d %H:%M:%S' | tee /test_result.txt
+LLVM_PROFILE_FILE='client_coverage_%5m.profraw' clickhouse-test -j "$(nproc)" --testname --shard --zookeeper --print-time 2>&1 | ts '%Y-%m-%d %H:%M:%S' | tee /test_result.txt
```
Comment (alesapin) [inline docker/test/coverage/run.sh:83]: image is abandoned and probably we have to remove it.

File: docker/test/fasttest/run.sh (hunk 1 above)
Comment (Felixoid) [reply docker/test/fasttest/run.sh:256]: Why do we limit it to 8? The host running fast tests has 32 CPUs on it

Comment (alesapin) [reply docker/test/fasttest/run.sh:256]: It's an amount of concurrent tests. One test can run query which utilizes all cores.

Comment (alexey-milovidov) [reply docker/test/fasttest/run.sh:256]: But usually don't.

For full CPU usage we can have `$(($(nproc) / 2))`
Maybe some tests will become flaky, but our flaky check should always cover most of these cases.

File: docker/test/coverage/run.sh (hunk 3 above)
Comment (Felixoid) [reply docker/test/coverage/run.sh]: Deleted

File: docker/test/fasttest/run.sh (hunk 1 above)
Comment (Felixoid) [reply docker/test/fasttest/run.sh]: I've done it as a half of CPUs with min 1

**Review by alesapin — APPROVED** [review]
Need to remove coverage image from docker/images.json.

**Comment by Felixoid** [comment]
Removed in the latest push, will merge today after the success running

**Pull request merged**